draft-ietf-tls-esni.md

```diff
@@ -763,6 +761,23 @@ If the server sends a NewSessionTicket message, the corresponding ECH PSK MUST
 be ignored by all other servers in the deployment when not negotiating ECH,
 including servers which do not implement this specification.
 
+## Backend Server Behavior {#backend-server-behavior}
+
+When the client-facing server accepts ECH, it forwards the ClientHelloInner to
+the backend server, who terminates the connection. If the ClientHelloInner
+contains an empty "encrypted_client_hello" extension, then the backend server
+MUST confirm ECH acceptance by setting ServerHello.random[24:32] to
+
+~~~~
+    accept_confirmation = HKDF-Expand-Label(
+        HKDF-Extract(0, ClientHelloInner.random),
+        "ech-eccept-confirmation",
+        ServerHello.random[0:24], 8)
+~~~~
+
+where HKDF-Expand-Label and HKDF-Extract are as defined in {{RFC8446}}. The
+value of ServerHello.random[0:24] is generated as usual by invoking a secure
+pseudorandom number generator.
```

Comment:
This doesn't seem part of this PR?

Reply:
This PR is based on #287, which is why this text appears here. If you focus your review on the last commit, it should be clear what the changes are.